cg/store/api/find_basic_data.py

```diff
@@ -159,23 +173,26 @@ def get_collaboration_by_internal_id(self, internal_id: str) -> Collaboration:
         ).first()
 
     def current_application_version(self, tag: str) -> Optional[ApplicationVersion]:
-        """Fetch the current application version for an application tag."""
-        application_obj = self.Application.query.filter_by(tag=tag).first()
-        if not application_obj:
+        """Return the current application version for an application tag."""
+        application = self.get_application_by_tag(tag=tag)
+        if not application:
             return None
-        application_id = application_obj.id
-        records = self.ApplicationVersion.query.filter_by(application_id=application_id)
-        records = records.filter(self.ApplicationVersion.valid_from < dt.datetime.now())
-        records = records.order_by(desc(self.ApplicationVersion.valid_from))
-
-        return records.first()
+        application_id = application.id
+        return apply_application_versions_filter(
+            filter_functions=[
+                ApplicationVersionFilter.FILTER_BY_APPLICATION_ID,
+                ApplicationVersionFilter.FILTER_BY_DATE,
+                ApplicationVersionFilter.ORDER_BY_VALID_FROM,
+            ],
+            application_versions=self._get_query(table=ApplicationVersion),
+            application_id=application_id,
+            date=dt.datetime.now(),
+        ).first()
 
     def latest_version(self, tag: str) -> Optional[ApplicationVersion]:
         """Fetch the latest application version for an application tag."""
-        application_obj = self.Application.query.filter_by(tag=tag).first()
-        return (
-            application_obj.versions[-1] if application_obj and application_obj.versions else None
-        )
+        application = self.get_application_by_tag(tag=tag)
+        return application.versions[-1] if application and application.versions else None
```

Comment:
What happens if the versions list is empty? Add a test case and maybe some error handling

Reply:
will include in another PR after refactoring this function